src/Domain/Block.cpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/Block.hpp"
+
+#include <ostream>
+
+#include "Utilities/StdHelpers.hpp"
+
+template <size_t VolumeDim>
+Block<VolumeDim>::Block(
+    const EmbeddingMap<VolumeDim, VolumeDim>& embedding_map, const size_t id,
+    std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>&&
+        neighbors)
+    : embedding_map_(embedding_map.get_clone()),
+      id_(id),
+      neighbors_(std::move(neighbors)) {
+  // Loop over Directions to search which Directions were not set to neighbors_,
+  // set these Directions to external_boundaries_.
+  for (size_t j = 0; j < 2 * VolumeDim; j++) {
+    // This is a map from [0,1,2,3,4,5] to [0,0,1,1,2,2], with 0-xi, 1-eta,
+    // 2-zeta:
+    size_t dimension = (j - j % 2) / 2;
+    Side side;
+    // With even/odd-ness designating lower/upper:
+    side = j % 2 == 1 ? Side::Upper : Side::Lower;
+    Direction<VolumeDim> input_direction(dimension, side);
+    if (neighbors_.find(input_direction) == neighbors_.end()) {
+      external_boundaries_.emplace(std::move(input_direction));
```

Comment:
@osheamonn is right as long as the fairly is there :) Whether or not you want to move depends on if you need to use the object after the move. If you do not then it's fine to move the object. In the case here we do not use `input_direction` after so it is safe to move.

Now the other part of the question is when do we _need_ to use `std::move`. Technically never, but let's instead ask when do we want to. If you have an lvalue reference and you no longer need it after that point and the function you're calling has an overload that takes a rvalue or forwarding reference you will see a performance benefit without causing undefined behavior.